CppLinq/cpplinq.hpp

```diff
@@ -3243,39 +3498,39 @@ namespace cpplinq
 
         // -------------------------------------------------------------------------
 
-        template<typename TKeyPredicate>
+        template<typename KeySelector>
```

Comment:
Predicate => Selector is fine (correct even) but in general template parameters are prefixed with 'T'. Any special reason to remove it here?